TreeDB/cmd/treedb_rag_interleave/main.go

```
@@ -0,0 +1,608 @@
+}
+
+func isRevision(value string) bool {
+	if len(value) != 40 {
+		return false
+	}
+	_, err := hex.DecodeString(value)
+	return err == nil
+}
+
+func validateProductAncestor(repo, productRevision, buildRevision string) error {
+	command := exec.Command("git", "-C", repo, "merge-base", "--is-ancestor", productRevision, buildRevision)
```

golangci-lint (2.12.2)

[medium] G204: Subprocess launched with variable

(gosec)